src/models/thread.ts

```diff
@@ -504,8 +504,42 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         throw new Error("Unsupported function on the thread model");
     }
 
+    public getEventReadUpTo(userId: string, ignoreSynthesized?: boolean): string | null {
+        const isCurrentUser = userId === this.client.getUserId();
+        if (isCurrentUser && this.lastReply()) {
+            // If a thread last activity is prior the first read receipt sent in a thread
+            // we want to consider that this thread has been read
+            const beforeFirstThreadedReceipt = (this.lastReply()?.getTs() ?? 0) < this.room.oldestThreadedReceiptTs;
+            if (beforeFirstThreadedReceipt) {
+                return this.timeline.at(-1)?.getId() ?? null;
+            }
+        }
+
+        const readUpToId = super.getEventReadUpTo(userId, ignoreSynthesized);
+
+        // Checking whether the unthreaded read receipt for that user is more recent
+        // than the read receipt inside that thread.
+        if (isCurrentUser && this.lastReply()) {
+            const unthreadedReceiptTs = this.room.unthreadedReceipts.get(userId)?.ts ?? Infinity;
+            for (let i = this.timeline?.length - 1; i >= 0; --i) {
+                const ev = this.timeline[i];
+                if (ev.getTs() > unthreadedReceiptTs) return ev.getId() ?? readUpToId;
+                if (ev.getId() === readUpToId) return readUpToId;
```

Comment:
Also I wonder if we should check the event ID against `readUpToId` first since timestamps are theoretically not 100% reliable?

I'm not sure this is correct -- I think you want `ev.getTs() <= unthreadedReceipTs` as we want to return the last read event? I think the current code will almost always return the latest event in the timeline? (I think this would then need to default to `0` if there is no unthreaded receipt, this better matches what happens in `hasUserReadEvent` though?)